### PR TITLE
Update to R4 RP standard for upcoming GA

### PR DESCRIPTION
**Pull request by leonardbf**
NetApp internal ticket NFSAAS-2345

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by niander** [comment]
@leonardbf It seems you have added new cmdlets. Have you started a new cmdlet design review? If yes please link it here.

**Comment by leonardbf** [comment]
@niander command review here https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/314

**Comment by leonardbf** [comment]
> @leonardbf It seems you have added new cmdlets. Have you started a new cmdlet design review? If yes please link it here.

@cormacpayne I see you were assigned to this. Can you advise perhaps on why there is a markdown failure here  https://dev.azure.com/azure-sdk/public/_build/results?buildId=40086. The message states there are no examples, however one is present. There is also a security message reported that I'm unsure how to resolve. Thanks.

**Comment by cormacpayne** [comment]
> 
> 
> > @leonardbf It seems you have added new cmdlets. Have you started a new cmdlet design review? If yes please link it here.
> 
> @cormacpayne I see you were assigned to this. Can you advise perhaps on why there is a markdown failure here https://dev.azure.com/azure-sdk/public/_build/results?buildId=40086. The message states there are no examples, however one is present. There is also a security message reported that I'm unsure how to resolve. Thanks.

@leonardbf I think it was an issue with how the validation tool checked for examples -- I removed an empty line that was inside of the example block that was probably causing the issue.

File: src/NetAppFiles/NetAppFiles/Account/NewNetAppFilesAccount.cs [hunk 1]
```diff
@@ -46,25 +45,32 @@ public class NewAzureRmNetAppFilesAccount : AzureNetAppFilesCmdletBase
...
             HelpMessage = "The name of the ANF account")]
         [ValidateNotNullOrEmpty]
         [Alias("AccountName")]
         public string Name { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable array which represents the active directories")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable[] ActiveDirectories { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Account/NewNetAppFilesAccount.cs]: What are the contets of this hashtable?  Also, parameter names should be singular 'ActiveDirectoryConfiguration' for example.

File: src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        DefaultParameterSetName = FieldsParameterSet), OutputType(typeof(PSNetAppFilesAccount))]
+    [Alias("Set-AnfAccount")]
+    public class SetAzureRmNetAppFilesAccount : AzureNetAppFilesCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter()]
+        public string ResourceGroupName { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs]: See: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/set-cmdlet-example.md

You must implement the InoutObject and ResourceId parameter sets as wekl.

File: src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            HelpMessage = "The name of the ANF account")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AccountName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable array which represents the active directories")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable[] ActiveDirectories { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs]: Same comments as above on this parameter

File: src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        public override void ExecuteCmdlet()
+        {
+            var netAppAccountBody = new NetAppAccount()
+            {
+                Location = Location,
+                ActiveDirectories = (ActiveDirectories != null) ? ModelExtensions.ConvertActiveDirectoriesFromPs(ActiveDirectories) : new List<ActiveDirectory>(),
+                Tags = Tag
+            };
+
+            if (ShouldProcess(Name, "Create the new account"))
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Account/SetNetAppFilesAccount.cs]: Provide details on the identity of the created account (resource id or account name)

File: src/NetAppFiles/NetAppFiles/Account/UpdateNetAppFilesAccount.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            HelpMessage = "The location of the resource")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.NetApp/netAppAccounts")]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable array which represents the active directories")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable[] ActiveDirectories { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Account/UpdateNetAppFilesAccount.cs]: What does this cmdlet allow you to change about the resource?  Also, same parameter name and parameter set comments

File: src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs [hunk 6]
```diff
@@ -97,12 +98,24 @@ public class NewAzureRmNetAppFilesVolume : AzureNetAppFilesCmdletBase
...
         [ValidateNotNullOrEmpty]
-        [PSArgumentCompleter("Standard", "Premium", "Extreme")]
+        [PSArgumentCompleter("Standard", "Premium", "Ultra")]
         public string ServiceLevel { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable array which represents the export policy")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable[] ExportPolicy { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs]: What is the schema for this parameter?  What kind of data do you expect users to provide?

File: src/NetAppFiles/NetAppFiles/Volume/UpdateNetAppFilesVolume.cs [hunk 7]
```diff
@@ -96,13 +98,27 @@ public class UpdateAzureRmNetAppFilesVolume : AzureNetAppFilesCmdletBase
         [ValidateNotNullOrEmpty]
         public string ServiceLevel { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable array which represents the export policy")]
+        [ValidateNotNullOrEmpty]
+        public Hashtable[] ExportPolicy { get; set; }
```
Comment (markcowl) [inline src/NetAppFiles/NetAppFiles/Volume/UpdateNetAppFilesVolume.cs]: What type of data does this contain?

**Comment by tomasm9** [comment]
I am not able to push my changes to this PR, some permission troubles... 

Is it okay if I create a new PR?

**Comment by markcowl** [comment]
@tomasm9 Feel free.  I will close this one, then, please reference this PR in the comments of your new PR.

**Comment by tomasm9** [comment]
https://github.com/Azure/azure-powershell/pull/9382 new pull request here

**Pull request closed without merging**